### PR TITLE
Refactored cont.index.R and created testcase

### LINKED ISSUE #111: Refactor `cont.index.R`
- [ ] Delete not used variables
- [ ] Extract functions
- [ ] Introduce tibbles

### PR DESCRIPTION
**Pull request by wverastegui**
Close #111

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,12 +11,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- added tests and testdata for run_filter.R (#155)
 ### Changed
 - refactored `feature.align.R` [#63](https://github.com/RECETOX/recetox-aplcms/pull/63)[#88](https://github.com/RECETOX/recetox-aplcms/pull/88)[#102](https://github.com/RECETOX/recetox-aplcms/pull/102)
 - refactored `adjust.time.R` [#64](https://github.com/RECETOX/recetox-aplcms/pull/64)[#102](https://github.com/RECETOX/recetox-aplcms/pull/102)
 - refactored `find.tol.time.R` [#91](https://github.com/RECETOX/recetox-aplcms/pull/91)
 - refactored `find.turn.point.R` [#91](https://github.com/RECETOX/recetox-aplcms/pull/91)
 - refactored `proc.cdf.R` and `adaptive.bin.R` [#137](https://github.com/RECETOX/recetox-aplcms/pull/137)
+- refactored `cont.index.R` and renamed as `run_filter.R` [#xx](https://github.com/RECETOX/recetox-aplcms/pull/155)
+- Updated function names `proc.cdf.R` and `extract_features.R` [#xx](https://github.com/RECETOX/recetox-aplcms/pull/155)
+- refactored `cont.index.R` and renamed as `run_filter.R` [#xx](https://github.com/RECETOX/recetox-aplcms/pull/155)
```
Comment (hechth) [inline CHANGELOG.md]:
```suggestion
- refactored `cont.index.R` and renamed as `run_filter.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
- Updated function names `proc.cdf.R` and `extract_features.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
- refactored `cont.index.R` and renamed as `run_filter.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,12 +11,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - added tests for `feature.align.R` ([#40](https://github.com/RECETOX/recetox-aplcms/pull/40)), and `adjust.time.R` ([#39](https://github.com/RECETOX/recetox-aplcms/pull/40))
 - added CI to repository's GitHub Actions [#45](https://github.com/RECETOX/recetox-aplcms/pull/45),[#49](https://github.com/RECETOX/recetox-aplcms/pull/49)
 - added additional test cases for hybrid [#133](https://github.com/RECETOX/recetox-aplcms/pull/133)
+- added tests and testdata for run_filter.R (#155)
```
Comment (hechth) [inline CHANGELOG.md]:
```suggestion
- added tests and testdata for run_filter.R (#156)
```

File: R/run_filter.R [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#' @param min_count_run filter parameter. 
+#' @param min_pres Run filter parameter. The minimum proportion of presence in the time period for a series of signals grouped
+#'  by m/z to be considered a peak.
+#' @param profile The matrix containing m/z, retention time, intensity, and EIC label as columns.
+#' @return unique_grp. 
+compute_uniq_grp <- function(profile, min_count_run, min_pres = 0.6) {
+  grps <- profile  
+  ttt <- table(grps)
+  ttt <- ttt[ttt >= max(min_count_run * min_pres, 2)]
+  unique_grp <- as.numeric(names(ttt))
```
Comment (hechth) [inline R/run_filter.R:12]: this can probably be simplified to not having to use the table function

File: tests/remote-files/run_filter.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+https://gitlab.ics.muni.cz/umsa/umsa-files/-/raw/master/testdata/recetox-aplcms/filtered/run_filter/mbr_test0_run_filter.parquet
```
Comment (hechth) [inline tests/remote-files/run_filter.txt:1]: You need to add the link to the other input file

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,12 +11,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - added tests for `feature.align.R` ([#40](https://github.com/RECETOX/recetox-aplcms/pull/40)), and `adjust.time.R` ([#39](https://github.com/RECETOX/recetox-aplcms/pull/40))
 - added CI to repository's GitHub Actions [#45](https://github.com/RECETOX/recetox-aplcms/pull/45),[#49](https://github.com/RECETOX/recetox-aplcms/pull/49)
 - added additional test cases for hybrid [#133](https://github.com/RECETOX/recetox-aplcms/pull/133)
+- added tests and testdata for run_filter.R (#156)
```
Comment (xtrojak) [inline CHANGELOG.md]:
```suggestion
- added tests and testdata for run_filter.R [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,12 +11,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Changed
 - refactored `feature.align.R` [#63](https://github.com/RECETOX/recetox-aplcms/pull/63)[#88](https://github.com/RECETOX/recetox-aplcms/pull/88)[#102](https://github.com/RECETOX/recetox-aplcms/pull/102)
 - refactored `adjust.time.R` [#64](https://github.com/RECETOX/recetox-aplcms/pull/64)[#102](https://github.com/RECETOX/recetox-aplcms/pull/102)
 - refactored `find.tol.time.R` [#91](https://github.com/RECETOX/recetox-aplcms/pull/91)
 - refactored `find.turn.point.R` [#91](https://github.com/RECETOX/recetox-aplcms/pull/91)
 - refactored `proc.cdf.R` and `adaptive.bin.R` [#137](https://github.com/RECETOX/recetox-aplcms/pull/137)
+- refactored `cont.index.R` and renamed as `run_filter.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
+- Updated function names `proc.cdf.R` and `extract_features.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
+- refactored `cont.index.R` and renamed as `run_filter.R` [#156](https://github.com/RECETOX/recetox-aplcms/pull/156)
```
Comment (xtrojak) [inline CHANGELOG.md]: Function names `proc.cdf` and `extract_features` were not changed + duplicated entry.

```suggestion
```

File: tests/testthat/test-run_filter.R [hunk 7]
```diff
@@ -0,0 +1,33 @@
+patrick::with_parameters_test_that(
+  "test run_filter.R",
```
Comment (xtrojak) [inline tests/testthat/test-run_filter.R]: To keep it consistent with other test names.

```suggestion
  "test run_filter",
```

File: R/run_filter.R [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      measured_points[this_times] <- 1
+
+      good_sel <- which(this_smooth >= min_pres)
+      good_points[good_sel] <- 1
+      for (j in (-min_run):min_run)
+      {
+        curr_sel <- good_sel + j
+        curr_sel <- curr_sel[curr_sel > 0 & curr_sel <= length(times)]
+        good_points[curr_sel] <- 1
+      }
```
Comment (xtrojak) [inline R/run_filter.R]: Just a minor style thing - consider keeping the curly brackets consistent (not just here). Maybe once you resolve the conflicts, R lint might have a comment on that (I'm not sure).

```suggestion
    if (max(this_smooth) >= min_pres) {
      measured_points <- good_points <- timeline
      measured_points[this_times] <- 1

      good_sel <- which(this_smooth >= min_pres)
      good_points[good_sel] <- 1
      for (j in (-min_run):min_run) {
        curr_sel <- good_sel + j
        curr_sel <- curr_sel[curr_sel > 0 & curr_sel <= length(times)]
        good_points[curr_sel] <- 1
      }
```

**Review by xtrojak — CHANGES_REQUESTED** [review]
Good job @wverastegui !

File: tests/testthat/test-run_filter.R [hunk 9]
```diff
@@ -0,0 +1,33 @@
+patrick::with_parameters_test_that(
+  "test run_filter",
+  {
+    if(ci_skip == TRUE) skip_on_ci()
+
+    testdata <- file.path("..", "testdata")
+    input_path <- file.path(testdata, "input", filename)
```
Comment (xtrojak) [inline tests/testthat/test-run_filter.R]: There are only `mzml` files in `input` directory. Maybe you meant `filtered`? This is related to CI error:

```
── Error ('test-run_filter.R:9'): test run_filter mbr_test0 ────────────────────
Error: IOError: Failed to open local file '../testdata/input/mbr_test0.parquet'. Detail: [errno 2] No such file or directory
 Backtrace:
    ▆
 1. ├─rlang::eval_tidy(code, args)
 2. ├─base::as.matrix(arrow::read_parquet(input_path)) at test-run_filter.R:9:4
 3. └─arrow::read_parquet(input_path)
 4.   └─arrow:::make_readable_file(file)
 5.     └─arrow::mmap_open(file)
 6.       └─arrow:::io___MemoryMappedFile__Open(path, mode)
```

Comment (wverastegui) [reply tests/testthat/test-run_filter.R]: The test file has been updated to the folder containing the input file (~/filtered/run_filter/). [7a81c10]

**Comment by hechth** [comment]
The reason why tests fail is because the testdata is already updated for merging the pr #154

**Pull request merged**